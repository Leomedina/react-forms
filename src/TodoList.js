import React, { useState } from 'react';
import { v4 as uuidv4 } from 'uuid';
import TodoForm from './TodoForm';


const TodoList = () => {
  const INITIAL_STATE = [
    { id: uuidv4(), task: "run" },
    { id: uuidv4(), task: "eat" }
  ];

  const [todos, setTodos] = useState(INITIAL_STATE);

  const addTodo = (newTodo) => {
    setTodos(todos => [...todos, { id: uuidv4(), ...newTodo }]);
  }

  const removeTodo = (id) => {
    const newArray = todos.filter(todo => todo.id !== id);
    setTodos(newArray);
    
  }
  return (
    <>
      <ul>
        {todos.map(({ id, task }) =>
          <>
            <li>{task}</li>
            <button onClick={() => removeTodo(id)}>X</button>
          </>
        )}
      </ul>
      <TodoForm addTodo={addTodo} />
    </>
  );
}

export default TodoList;