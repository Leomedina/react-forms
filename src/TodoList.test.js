import React from "react";
import { render } from "@testing-library/react";
import TodoList from './TodoList';

describe("Smoke and snapshot tests", function () {
  it("should render without crashing", function () {
    render(<TodoList />);
  });

  it("should match snapshot", function () {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
  });
});