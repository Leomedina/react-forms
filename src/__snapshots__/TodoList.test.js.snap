// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <ul>
    <li>
      run
    </li>
    <button>
      X
    </button>
    <li>
      eat
    </li>
    <button>
      X
    </button>
  </ul>
  <form>
    <label
      for="task"
    >
      Todo:
    </label>
    <input
      id="task"
      name="task"
      type="text"
      value=""
    />
    <button>
      Add
    </button>
  </form>
</DocumentFragment>
`;
